src/Simplex/FileTransfer/Agent.hs

```diff
@@ -214,6 +217,19 @@ runXFTPRcvWorker c srv Worker {doWork} = do
       when complete . lift . void $
         getXFTPRcvWorker True c Nothing
       where
+        unknownNoProxy :: AM Bool
+        unknownNoProxy = do
+          let AgentClient {xftpServers, useNetworkConfig} = c
+          knownSrvs_ <- atomically $ TM.lookup userId xftpServers
+          let srvKnown = maybe False (\knownSrvs -> srv `elem` L.map protoServer knownSrvs) knownSrvs_
+          -- assuming specific XFTP client will have the same socksProxy config
+          (_, NetworkConfig {socksProxy, hostMode, requiredHostMode}) <- readTVarIO useNetworkConfig
+          let usingProxy =
+                -- on Android we have direct information about using proxy
+                isJust socksProxy
+                  -- on iOS we can infer based on hostMode and requiredHostMode
+                  || ((hostMode == HMOnionViaSocks || hostMode == HMOnion) && requiredHostMode)
+          pure $ not srvKnown && not usingProxy
```

Comment:
it may make sense to separate this into new agent api to not repeat logic in chat and UI - similar to `connectionPlan` - `checkUnknownNoProxy`

Reply:
It is better to decompose it differently.

1) whether the file has approved relays - this check should be done in the chat (and can be exposed via the API), at the time that the decision to download is made in the UI. That sets the parameter on the file whether it is approved relays (either because user approved them or because they are in approved list).
2) is current connection via Tor - this would happen both in UI and in the agent. 

This way the change in known servers won't affect the outcome at the time of the check - e.g. user removing known proxies from the list shouldn't prevent completion of the download.

The agent would then only stop download if file relays are not approved.

Also, the term we should use is socksProxy, not just proxy, or it'll be confused with protocol proxies.